src/combinations.rs

```diff
@@ -210,8 +209,17 @@ where
 {
 }
 
+pub(crate) fn n_and_count<I: Iterator>(
```

Comment:
Can you provide documentation here?

Reply:
Added documentation, although as far as I can tell the existence of that function (and not just `count` by itself) is to enable `powerset::count` so maybe there is a better refactoring here.

Also added the specialization tests, but I'm not really sure what that's doing so adapted the one for `tuple_combinations`.